### ISSUE TITLE
final escaped field value lost when trailing newline is absent

### ISSUE THREAD

**Issue by superhawk610** [beatrichartz/csv#117]
When the final value on the final line of a file without a trailing newline is escaped (wrapped in quotes), it's dropped from the output and an empty string is returned instead. For example:

```csv
Budget
$2500.00
```

will output `%{"Budget" => "$2500.00"}`, but

```csv
Budget
"$2500.00"
```

will output `%{"Budget" => ""}`.

Note that this only occurs when the trailing newline is omitted from the file, regardless of line endings (LF or CRLF both exhibit this behaviour). If the trailing newline is present, this behaviour does not occur.

I would expect `"$2500.00"` to be returned for `Budget` in both cases above, regardless of whether it's wrapped in escape characters or not.

## Steps to Reproduce

```elixir
parse_csv = fn path ->
  path
  |> Path.expand()
  |> File.stream!([:utf8])
  |> CSV.decode!(headers: true)
  |> Enum.to_list()
end

parse_csv.("~/Downloads/with-quotes.csv")
#=> [%{"Budget" => ""}]

parse_csv.("~/Downloads/without-quotes.csv")
#=> [%{"Budget" => "$2500.00"}]
```

## CSV Files

[with-quotes.csv](https://github.com/beatrichartz/csv/files/9914337/with-quotes.csv)
[without-quotes.csv](https://github.com/beatrichartz/csv/files/9914339/without-quotes.csv)

**Comment by superhawk610** [comment]
This issue was introduced in `3.0.0` - I cannot reproduce it on `2.5.0`.

**Comment by superhawk610** [comment]
It seems this may be related to #112. If it's much effort to fix, just adding a note in the changelog/readme that a trailing newline is required would be helpful.

**Comment by beatrichartz** [comment]
Thanks for this - does this still happen in `3.0.1`? Only checking since you mentioned `3.0.0` and some related issues have been fixed in `3.0.1`

**Comment by beatrichartz** [comment]
I was able to reproduce this, will release the fix in `3.0.2`. Thanks again! Feel free to reopen if the issue persists.
